--- src/nb_func_orig.cpp	2026-05-18 10:28:14.994072323 -0500
+++ src/nb_func.cpp	2026-05-18 10:28:32.076587784 -0500
@@ -18,6 +18,10 @@
 #  include <cxxabi.h>
 #endif

+#if defined(_AIX) && defined(func_data)
+#undef func_data
+#endif
+
 #if defined(_MSC_VER)
 #  pragma warning(disable: 4706) // assignment within conditional expression
 #  pragma warning(disable: 6255) // _alloca indicates failure by raising a stack overflow exception

--- src/nb_internals_orig.h	2026-05-18 10:12:39.850505790 -0500
+++ src/nb_internals.h	2026-05-18 10:12:46.870504034 -0500
@@ -21,6 +21,10 @@
 #include <functional>
 #include "hash.h"

+#if defined(_AIX) && defined(func_data)
+#undef func_data
+#endif
+
 #if TSL_RH_VERSION_MAJOR != 1 || TSL_RH_VERSION_MINOR < 3
 #  error nanobind depends on tsl::robin_map, in particular version >= 1.3.0, <2.0.0
 #endif

--- src/nb_internals_orig.cpp	2026-05-18 10:12:09.312772129 -0500
+++ src/nb_internals.cpp	2026-05-18 10:12:16.285096192 -0500
@@ -13,6 +13,10 @@
 #include "nb_abi.h"
 #include <thread>

+#if defined(_AIX) && defined(func_data)
+#undef func_data
+#endif
+
 #if defined(__GNUC__) && !defined(__clang__)
 #  pragma GCC diagnostic ignored "-Wmissing-field-initializers"
 #endif

--- cmake/nanobind-config_orig.cmake	2026-05-18 11:33:10.531870221 -0500
+++ cmake/nanobind-config.cmake	2026-05-18 11:26:20.930507133 -0500
@@ -207,7 +207,7 @@
     nanobind_lto(${TARGET_NAME})

     nanobind_strip(${TARGET_NAME})
-  elseif(NOT WIN32 AND NOT APPLE)
+  elseif(NOT WIN32 AND NOT APPLE AND NOT AIX)
     target_compile_options(${TARGET_NAME} PUBLIC $<${NB_OPT_SIZE}:-ffunction-sections -fdata-sections>)
     target_link_options(${TARGET_NAME} PUBLIC $<${NB_OPT_SIZE}:-Wl,--gc-sections>)
   endif()
@@ -307,6 +307,9 @@
 endfunction()

 function (nanobind_lto name)
+  if (CMAKE_SYSTEM_NAME MATCHES "AIX")
+    return()
+  endif()
   set_target_properties(${name} PROPERTIES
     INTERPROCEDURAL_OPTIMIZATION_RELEASE ON
     INTERPROCEDURAL_OPTIMIZATION_MINSIZEREL ON)

--- include/nanobind/nb_call_orig.h	2026-05-24 07:19:48.602347481 -0500
+++ include/nanobind/nb_call.h	2026-05-24 07:20:32.628186792 -0500
@@ -15,6 +15,10 @@
 #  pragma warning(disable: 6255) // _alloca indicates failure by raising a stack overflow exception
 #endif

+#if defined(_AIX)
+#include <alloca.h>
+#endif
+
 class kwargs_proxy : public handle {
 public:
     explicit kwargs_proxy(handle h) : handle(h) { }

--- src/nb_ft_orig.cpp	2026-05-24 07:24:39.565601284 -0500
+++ src/nb_ft.cpp	2026-05-24 07:24:57.041888001 -0500
@@ -12,6 +12,10 @@
     op->ob_ref_shared = 0;
 }

+#if defined(_AIX)
+#include <alloca.h>
+#endif
+
 #if PY_VERSION_HEX < 0x030E00A5
 void nb_enable_try_inc_ref(PyObject *obj) noexcept {
     // Since this is called during object construction, we know that we have
